python/ray/autoscaler/_private/vsphere/pyvmomi_sdk_provider.py

```diff
@@ -37,22 +59,18 @@ def get_pyvmomi_obj_by_moid(self, vimtype, moid):
         if self.pyvmomi_sdk_client is None:
             raise ValueError("Must init pyvmomi_sdk_client first.")
 
-        container = self.pyvmomi_sdk_client.viewManager.CreateContainerView(
-            self.pyvmomi_sdk_client.rootFolder, vimtype, True
+        if not moid:
+            raise ValueError("Invalid argument for moid")
+
+        container = self.pyvmomi_sdk_client.content.viewManager.CreateContainerView(
+            self.pyvmomi_sdk_client.content.rootFolder, vimtype, True
         )
 
         for c in container.view:
-            if moid:
-                if moid in str(c):
-                    obj = c
-                    break
-            else:
+            if moid in str(c):
                 obj = c
                 break
-        if not obj:
-            raise RuntimeError(
-                f"Unexpected: cannot find vSphere object {vimtype} with moid: {moid}"
-            )
```

Comment:
- What's the reason for the changes to this function?
- Previously `get_pyvmomi_obj_by_*` could not return `None` (instead it would raise an exception), but now it can return None.  Do the callers need to handle the case where it's `None`?

Reply:
- I think the reason is to simplify the function's logic. Actually when I do the cherry pick for the bug fix, I find that this file has been changed in another PR by some other teammate. So I also cherry-picked this commit.
- Yes, I agree with you that we should never change the behaviour like this when no change to the callers of this function. So let me optimize the function to make sure it will throw an exception when things are not correct.